tests/trainer/test_trainer.py

```diff
@@ -276,7 +278,8 @@ def mock_save_function(filepath, *args):
     # simulated losses
     losses = [10, 9, 2.8, 5, 2.5]
 
-    checkpoint_callback = ModelCheckpoint(tmpdir, save_top_k=save_top_k, prefix=file_prefix, verbose=1)
+    checkpoint_callback = ModelCheckpoint(tmpdir, save_top_k=save_top_k, save_last=save_last,
+                                          prefix=file_prefix, verbose=1)
     checkpoint_callback.save_function = mock_save_function
```

Comment:
does this test actually check that the last checkpoint is saved?
it seems it only tests that there are k models when save_top_k is used.

Reply:
It checks but rather indirectly by doing `assert len(file_lists) == len(expected_files)`.

Do you think it's a better idea to check directly if `last.ckpt` exists? In this case we would also need to do something like `assert (len(file_lists) - (1 if save_last else 0)) == len(expected_files)` and not pass 'last.ckpt' to `expected_files`

I personally think the first option is easier to understand, but you might have some better ideas =)

Reply:
sorry, just saw that you added the extra params to the test with the "last.ckpt" file. Then I think the test is ok, but now I'm thinking "last" may not be a good file name because it doesn't tell us what the epoch number is. 
is there a reason why we can't call it the same as the other file names including the number?

Reply:
I think a constant name is good because it makes it easy for the user to always load the most trained model without having to keep track on how many epochs the model was trained on.

Of course the user could always iterate over the directory and pick the model with the highest epoch count, but I think that introduces unnecessary overhead.

Reply:
true, that's a good argument.